pkgs/by-name/fi/fim/package.nix

```diff
@@ -35,22 +36,31 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-/p7bjeZM46DJOQ9sgtebhkNpBPj2RJYY3dMXhzHnNmg=";
   };
 
+  patches = [
+    # build tools with a build compiler
+    ./native-tools.patch
```

Comment:
the better way to do this is to use `@cc_for_build@` in the .patch file and then substitute here instead of in `postPatch`:
```nix
patches = [
  (substituteAll {
    src = ./native-tools.patch;
    cc_for_build = lib.getExe buildPackages.stdenv.cc;
  })
];
```

see pkgs/by-name/gn/gnome-shell/package.nix for an example